src/HeadlessH5PServiceProvider.php

```diff
@@ -32,11 +34,10 @@ class HeadlessH5PServiceProvider extends ServiceProvider
 {
     public function register(): void
     {
-        $this->app->singleton(H5PCore::class, function ($app) {
-            return new HeadlessH5P();
-        });
         $this->commands([StorageH5PLinkCommand::class]);
         $this->bindH5P();
+
+        $this->app->bind(H5PContentRepositoryContract::class, H5PContentRepository::class);
```

Comment:
Sometimes I wonder - aren't all our Repositories Singletons? Wouldn't it be better to bind them using `app->singleton` method, so they are reused after first time? There should be no side-effects to that, as we don't keep any state in Repositories, right?